rotary/rules/rotary.smk

```diff
@@ -1003,29 +1022,29 @@ rule circularize:
 # TODO - can I auto-predict genome completeness, names, types, topologies?
 rule run_dfast:
     input:
-        contigs="circularize/circularize.fasta",
+        contigs="{sample}/circularize/circularize.fasta",
         install_finished=os.path.join(DB_DIR_PATH,"checkpoints","dfast_" + VERSION_DFAST)
     output:
-        "annotation/dfast/genome.fna",
-        "annotation/dfast/protein.faa"
+        dfast_genome = "{sample}/annotation/dfast/genome.fna",
+        dfast_proteins = "{sample}/annotation/dfast/protein.faa",
+        outdir = directory("{sample}/annotation/dfast")
     conda:
         "../envs/annotation_dfast.yaml"
     log:
-        "logs/annotation/annotation_dfast.log"
+        "{sample}/logs/annotation/annotation_dfast.log"
     benchmark:
-        "benchmarks/annotation/annotation_dfast.txt"
+        "{sample}/benchmarks/annotation/annotation_dfast.txt"
     params:
-        outdir="annotation/dfast",
         db=directory(os.path.join(DB_DIR_PATH,"dfast_" + VERSION_DFAST)),
-        strain=SAMPLE_ID
+        strain=lambda wildcards: wildcards.sample # Get sample name from wildcards.
```

Comment:
I'm not sure if this will work, but is it possible to simplify this code and just use `{sample}`? See https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#non-file-parameters-for-rules

Reply:
@jmtsuji Thanks. I misunderstood this. I thought you couldn't use written wildcards in params because of some of the other examples I was seeing.